Task6/task8.js

```diff
@@ -0,0 +1,23 @@
+function f(x) {
+    return Math.random()*x;
+}
+  
+function makeCaching(f) {
+    var cache = {};
+
+    return function(item) {
+        if (!(item in cache)) {
+            cache[item] = f.call(this, item);
+        }
+        return cache[item];
+    };
+}
+  
+f = makeCaching(f);
```

Comment:
Please, never use such assignments. You might accidentally overwrite something that you need.